subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -670,6 +678,49 @@ private void captureBuildOperationResult(BuildOperationContext context, Resolver
         });
     }
 
+    @Override
+    public ConfigurationInternal getConsistentResolutionSource() {
+        return consistentResolutionSource;
+    }
+
+    private void maybeConfigureConsistentResolution() {
+        if (consistentResolutionSource != null) {
+            assertThatConsistentResolutionIsPropertyConfigured();
+            consistentResolutionSource.getIncoming().getResolutionResult().allComponents(this::registerConsistentResolutionConstraint);
+        }
+    }
+
+    private void assertThatConsistentResolutionIsPropertyConfigured() {
```

Comment:
I think this should be called by `resolveConsistentlyWith` when setting the source and eagerly fail

Reply:
We can't because of laziness: the `canBeResolved` flag can be set _after_ this method is called.